asyncy/utils/HttpUtils.py

```
@@ -29,9 +29,11 @@ def read_response_body_quietly(response):
                 return res
             except HTTPError as e:
                 last_exception = e
-                logger.error(
-                    f'Failed to call {url}; attempt={attempts}; err={str(e)}'
-                )
+                if logger is not None:
```

Why will logger be none? This should be avoided at all costs, since we want logging for every failed HTTP call to go somewhere (be it user logs, or engine logs).

---

Better safe than sorry to ensure that there's an actual logger passed along in this method.

---

Na, this should __crash__ the engine directly and loudly as it would be a big programming error in the engine. We shouldn't program so defensively.

---

I believe if we're aiming for a stable production-ready platform, coding defensively is exactly what we need to do. Especially if enterprise readiness is part of our roadmap.

---

What happens in this particular scenario:
1. All logs are lost - lost executions, lost behaviour
2. 100% silent. Nobody is alerted. At least, if the logger throws an exception, we're notified of it instantly (if stories.storyscriptapp.com goes down, I react to it instantly)

---

Also, in the reporting system, I suppress log messages from http callbacks because I don't think the engine needs to crash if a reporting doesn't receive an event.

---

If you need to ignore an exception, simply consume it blindly:
```py
try:
  raise Exception()
except:
  pass
```